addons/mail/static/src/discuss/core/common/discuss_core_common_service.js

```diff
@@ -37,6 +37,12 @@ export class DiscussCoreCommon {
                     type: "info",
                 });
             }
+            thread.delete();
+            const newThread =
+                this.store.discuss.channels.threads.find(
+                    (thread) => thread.displayToSelf || thread.isLocallyPinned
+                ) || this.store.inbox;
+            newThread.setAsDiscussThread();
```

Comment:
Fix is incorrect: when leaving the conversation, this doesn't semantically means the thread doesn't exist.
Removing it from local knowledge is fine for reducing memory load, but it shouldn't be done for a semblance of correctness when the thread is not actually deleted server-side.

The `channel.markAsFetched()` comes from`"discuss.channel/new_message"` notification. This shouldn't be received and/or handled when the user has left. That's the part that should be fixed.

The `newThread.setAsDiscussThread()` part looks out-of-scope of PR. This looks wrong too: we can leave a conversation from discuss sidebar, so the thread is not necessarily the active one. It's not okay to set another `discuss.thread` when the thread that has been left is not the active thread.

Would also be nice to make a test to cover this problem.